### PR TITLE
Gravatar join images

### PR DESCRIPTION
**Pull request by tlehman**
Add `Hitch::Assets` to handle downloading gravatar images and merging them.

To do list:

 - [ ] Allow any number of users to merge their images (for example, `hitch tlehman therubymug poblahblahblah ronnycarr` would successfully merge all four avatars
 - [ ] Use RMagick *or* switch to pure Ruby library like [chunky PNG](https://github.com/wvanbergen/chunky_png)

### CODE REVIEW

**Comment by therubymug** [comment]
@tlehman it's not passing on TravisCI. :-/

File: lib/hitch/assets.rb [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+      def xor_email_md5s(email1, email2)
+        (md5(email1).to_i(16)^md5(email2).to_i(16)).to_s(16)
+      end
+
+      def merge_avatars_for_pair!(email1, email2)
+        filename1 = File.join(gravatar_images_path, md5(email1))
+        filename2 = File.join(gravatar_images_path, md5(email2))
+        filename1xor2 = File.join(gravatar_images_path, xor_email_md5s(email1, email2))
+
+        Kernel.system("convert #{filename1} #{filename2} +append #{filename1xor2}")
```
Comment (stephenprater) [inline lib/hitch/assets.rb]: You're implicitly depending on ImageMagick being present on the system here.  Instead you should add MiniMagick to the Gemfile and use the ruby interface.

File: spec/hitch/assets_spec.rb [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+  describe ".avatar_for_pair" do
+    context "when there is no merged gravatar image for pair" do
+      let(:md5_bender) { Digest::MD5.hexdigest(bender_email) }
+      let(:md5_zoidberg) { Digest::MD5.hexdigest(zoidberg_email) }
+      let(:filename_bender) { File.join(gravatar_images_path, "#{md5_bender}") }
+      let(:filename_zoidberg) { File.join(gravatar_images_path, "#{md5_zoidberg}") }
+
+      # xor is used to combine the md5s because it is commutative
+      let(:md5_bender_xor_md5_zoidberg) { File.join(gravatar_images_path, (md5_zoidberg.to_i(16)^md5_bender.to_i(16)).to_s(16)) }
```
Comment (stephenprater) [inline spec/hitch/assets_spec.rb]: That is clever.  Well done sir.

File: lib/hitch/assets.rb (hunk 1 above)
Comment (tlehman) [inline lib/hitch/assets.rb]: Good idea

File: spec/hitch/assets_spec.rb (hunk 2 above)
Comment (tlehman) [inline spec/hitch/assets_spec.rb]: Thank you

**Comment by tlehman** [comment]
@therubymug I noticed its only under ruby 1.8.7, should we even have a job for a version that's no longer supported?

**Comment by therubymug** [comment]
@tlehman good question. I'll make a commit that removes `1.8.7` and then you can pull those changes into your fork.

**Comment by therubymug** [comment]
@tlehman rebase your branch onto `master` it should make your for green now. :-)

**Comment by tlehman** [comment]
closed as not practical

**Pull request closed without merging**